extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -476,10 +484,25 @@ class MongoQueueProcessor {
         });
     }
 
+    _sendMetrics() {
+        Object.keys(this._accruedMetrics).forEach(loc => {
+            const count = this._accruedMetrics[loc];
+
+            if (count > 0) {
+                this._accruedMetrics[loc] -= count;
```

Comment:
since this method is called on a separate schedule from the processing of kafka entries, I wanted to guarantee that I didn't reset possible new metrics accrued

Reply:
I was just thinking about the logic of the computation, since you substract `count` which is the original counter value, the only final value you can get is 0. It would be simpler to just reset the counter to 0 then.

Reply:
The possibility of the counter changing in between would only be valid if different threads could change this value in parallel, but it's not the case (all javascript code executes in a single thread).

Reply:
ah! yea oops, will make this change, ty 👍